From: Andrew Morton <akpm@linux-foundation.org>
Subject: highmem-fix-kmap_to_page-for-kmap_local_page-addresses-fix
Date: Mon Oct 10 01:12:33 PM PDT 2022

fix warning

mm/highmem.c:164:7: warning: Local variable 'i' shadows outer variable [shadowVariable]

Reported-by: kernel test robot <lkp@intel.com>
Cc: Al Viro <viro@zeniv.linux.org.uk>
Cc: Christoph Hellwig <hch@lst.de>
Cc: "Fabio M. De Francesco" <fmdefrancesco@gmail.com>
Cc: Ira Weiny <ira.weiny@intel.com>
Cc: kernel test robot <lkp@intel.com>
Cc: Thomas Gleixner <tglx@linutronix.de>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/highmem.c |    7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

--- a/mm/highmem.c~highmem-fix-kmap_to_page-for-kmap_local_page-addresses-fix
+++ a/mm/highmem.c
@@ -160,11 +160,8 @@ struct page *__kmap_to_page(void *vaddr)
 
 	/* kmap() mappings */
 	if (WARN_ON_ONCE(addr >= PKMAP_ADDR(0) &&
-			 addr < PKMAP_ADDR(LAST_PKMAP))) {
-		int i = PKMAP_NR(addr);
-
-		return pte_page(pkmap_page_table[i]);
-	}
+			 addr < PKMAP_ADDR(LAST_PKMAP)))
+		return pte_page(pkmap_page_table[PKMAP_NR(addr)]);
 
 	/* kmap_local_page() mappings */
 	if (WARN_ON_ONCE(base >= __fix_to_virt(FIX_KMAP_END) &&
_
